src/main.c

```diff
@@ -0,0 +1,110 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "math.h"
+#include "debug.h"
+#include "helpers.h"
+
+
+int main(int argc, char *argv[]) {
+    int opt, num1, num2 = 0;            // Only works for integers for now
+    char operation = 0;             // Which operator we are going to use
+    int has_numbers = 0;            // Status: Does *argv have numbers?
+
+    // Parse command-line options
+    while ((opt = getopt(argc, argv, "a:s:m:d:hv")) != -1) // CMD Line options 
+    {
+        switch (opt) 
+        {
+            case 'a': // Add
+                operation = 'a';
+                num1 = atof(optarg);
+                num2 = atof(argv[optind]);
+                has_numbers = 1;
+                break;
+
+            case 's': // Subtract
+                operation = 's';
+                num1 = atof(optarg);
+                num2 = atof(argv[optind]);
+                has_numbers = 1;
+                break;
+
+            case 'm': // Multiply
+                operation = 'm';
+                num1 = atof(optarg);
+                num2 = atof(argv[optind]);
+                has_numbers = 1;
+                break;
+
+            case 'd': // Divide
+                operation = 'd';
+                num1 = atof(optarg);
+                num2 = atof(argv[optind]);
+                has_numbers = 1;
+                break;
+
+            case 'h': // Help
+                print_help();
+                return 0;
+
+            case 'v': // Version
+                print_version();
+                return 0;
+
+            default: // Invalid option
+                print_help();
+                return 1;
+        }
+    }
+
+    // Perform the selected operation
+    if (has_numbers) 
```

Comment:
Do you need this check if you are already returning early from help and print version?